runtime-spec/runtime-spec-transaction.tex

```diff
@@ -0,0 +1,333 @@
+    result in a valid signature if the extrinsic is within the specified period 
+    or an invalid signature if the extrinsic "died".
+
+    \subsubsection*{Example}
+
+    The extrinsic author choses $M_{per} = 256$ at block \verb|10'000|,
+    resulting with $M_{pha} = 16$. The extrinsic is then valid for blocks
+    ranging from \verb|10'000| to \verb|10'256|.
+
+    \subsubsection*{Encoding}\label{sect-mortality-encoding}
+
+    $T_m$ refers to the SCALE encoded form of type $M_{per}$ and $M_{pha}$.
+    $T_m$ is the size of two bytes if the extrinsic is considered mortal,
+    or simply one bytes with the value equal to zero if the extrinsic is
+    considered immortal.
+
+    \[
+        T_m := Enc_{SC}(M_{per}, M_{pha})
+    \]
+
+    The SCALE encoded representation of mortality $T_m$ deviates from most
+    other types, as it's specialized to be the smallest possible value, as
+    described in Algorithm \ref{algo-encode-mortality} and
+    \ref{algo-decode-mortality}.
+
+    \begin{algorithm}[H]
+        \caption[]{\sc Encode Mortality}
+        \label{algo-encode-mortality}
+        \begin{algorithmic}[1]
+            \Require{$M_{per}, M_{pha}$}
+            \Statex // If the extrinsic is immortal, specify
+            \Statex // a single byte with the value equal to zero.
+            \State $\vcenter{
+                \begin{flalign*}
+                    \Return & 
+                    \begin{cases}
+                    0 & if\ extrinsic\ is\ immortal 
+                    \end{cases}&
+                \end{flalign*}
+            }$
+            \State \textbf{Init} $factor = \textsc{Max}(M_{per} >> 12, 1)$
+            \State \textbf{Init} $left = \textsc{Min}(\textsc{Max}(\textsc{TZ}(M_{per})-1,\ 1),\ 15)$
+            \State \textbf{Init} $right = \frac{M_{pha}}{factor} << 4$
+            \Statex
+            \Statex // Returns a two byte value
+            \State \Return $left|right$
+        \end{algorithmic}
+    \end{algorithm}
+
+    \begin{algorithm}[H]
+        \caption[]{\sc Decode Mortality}
+        \label{algo-decode-mortality}
+        \begin{algorithmic}[1]
+            \Require{$T_m$}
+            \State $\vcenter{
+                \begin{flalign*}
+                    \Return & 
+                    \begin{cases}
+                    \textit{Immortal} & if\ T^{b0}_{mort} = 0
+                    \end{cases}&
+                \end{flalign*}
+            }$
+            \Statex 
+            \State \textbf{Init} $enc = T^{b0}_{mort} + (T^{b1}_{mort} << 8)$
+            \State \textbf{Init} $M_{per} = 2 << (enc\ mod\ (1 << 4))$
+            \State \textbf{Init} $factor = \textsc{Max}(M_{per} >> 12, 1)$
+            \State \textbf{Init} $M_{pha} = (enc >> 4) * factor$
+            \State $\vcenter{
+                \begin{flalign*}
+                    \Return & 
+                    \begin{cases}
+                    (M_{per}, M_{pha}) & if\ M_{per} \geq 4\ \textbf{and}\ M_{pha} < M_{per} \\
```

Comment:
It is probably in the code like this, but I would move these requirements up into the definition. That the phase is smaller then the period should is a hard requirement. And the encoding is not able to encode periods < 4 either.